tests/testsuite/git.rs

```diff
@@ -757,13 +757,11 @@ fn update_with_shared_deps() {
         .with_status(101)
         .with_stderr(
             "\
+[UPDATING] git repository [..]
 [ERROR] Unable to update [..]
 
 Caused by:
-  precise value for git is not a git revision: 0.1.2
-
-Caused by:
-  unable to parse OID - contains invalid characters; class=Invalid (3)
+  revspec '0.1.2' not found; class=Reference (4); code=NotFound (-3)
```

Comment:
> When used with spec, allows you to specify a specific version number to set the package to. If the package comes from a git repository, this can be a git revision (such as a SHA hash or tag).

The documentation only says SHA and doesn't specify if shorts are allowed.

I almost wonder if the reference to `tag` is the bug, especially when I see errors like this...

Reply:
It worked in 1.45.0 before #8363 merged and [was documented since then](https://doc.rust-lang.org/1.45.0/cargo/commands/cargo-update.html). The doc mentions [git revisions](https://git-scm.com/docs/revisions), which could be like anything. This could be considered as an old regression that nobody reported until recent.

Reply:
BTW the test was added to prevent panic when revision is missing: <https://github.com/rust-lang/cargo/pull/2097/commits/1ad6f785ca1f32871666497d437183c79c32ebc6>.